pkg/health/server/prober.go

```diff
@@ -75,7 +75,7 @@ func (p *prober) copyResultRLocked(ip string) *models.PathStatus {
 	}
 	for res, value := range paths {
 		if value != nil {
-			*res = &*value
```

Comment:
Your changes are correct but I believe this might be a bug. @joestringer is your memory fresh enough to understand what was the intent here?

Reply:
Just lurking here, but the code in that file is...interesting :raised_eyebrow: . I'm curious to hear what @joestringer has on this.

Reply:
I am also curious about the rationale behind this as well :smiley:

Reply:
Looks like an attempt to make a deep copy.

I recall being concerned about the common `p.results` being modified, and not wanting to unintentionally share the pointer `p.results[ip].Icmp` which could be modified, rather to take a 'copy' of the underlying value.
That said, I think both prior to this change and after this change we are effectively copying a pointer from `p.results[ip].Icmp` (for instance) into `result.Icmp` so if the core `p.results` update code modifies the structure in-place then it's broken either way. If the core update code creates a new structure and updates the pointer then we're fine.